### PR TITLE
Rasa extra env vars

### PR DESCRIPTION
**Pull request by whitedr**
This change adds a new value: `.Values.rasa.extraEnvs` which will set extra env vars in both rasa/rasa-x deployments.  The motivation for wanting this is perhaps best explained with an example.

```
rasa:
  additionalChannelCredentials: >
    twilio:
      account_sid: ${TWILIO_ACCOUNT_SID}
      auth_token: ${TWILIO_AUTH_TOKEN}
      twilio_number: ${TWILIO_NUMBER}
  extraEnvs:
    - name: "TWILIO_ACCOUNT_SID"
      valueFrom:
        secretKeyRef:
          name: twilio-auth
          key: twilio_account_sid
    - name: TWILIO_AUTH_TOKEN
      valueFrom:
        secretKeyRef:
          name: twilio-auth
          key: twilio_auth_token
    - name: TWILIO_NUMBER
      valueFrom:
        secretKeyRef:
          name: twilio-auth
          key: twilio_number
```

In this case, I have defined a k8s secret named `twilio-auth` and I want to set env vars based on keys in that secret and then use those env vars in the `.Values.rasa.additionalChannelCredentials`.  Allowing extraEnvs to be set on rasa/rasa-x allows such a use case.

Note that another use case it helps is to set the url of duckling from an env variable as that url is configured in `config.yaml` and I couldn't find another way to dynamically set that other than using an env variable (which thankfully rasa supports!).

### CODE REVIEW

File: charts/rasa-x/values.yaml [hunk 1]
```diff
@@ -67,6 +67,8 @@ rasa:
   initialProbeDelay: 10
   # lockStoreDatabase is the database in redis which Rasa uses to store the conversation locks
   lockStoreDatabase: "1"
+  # extra env variables
+  extraEnvs:
```
Comment (wochinge) [inline charts/rasa-x/values.yaml]: Is this supposed to be a string or a dict / array?

File: charts/rasa-x/values.yaml [hunk 2]
```diff
@@ -67,6 +67,8 @@ rasa:
   initialProbeDelay: 10
   # lockStoreDatabase is the database in redis which Rasa uses to store the conversation locks
   lockStoreDatabase: "1"
+  # extra env variables
```
Comment (wochinge) [inline charts/rasa-x/values.yaml]: let's follow the Helm best practices and start the comment with the same name as the variable:
```suggestion
  # extraEnvs are environment variables which can be added to the Rasa X deployment
```

File: charts/rasa-x/values.yaml (hunk 1 above)
Comment (wochinge) [reply charts/rasa-x/values.yaml]: I'd suggest to give a commented out example how this value should look like

File: charts/rasa-x/templates/rasa-deployments.yaml [hunk 3]
```diff
@@ -100,7 +100,7 @@ spec:
           value: "{{ .rasaEnvironment }}"
         - name: "RASA_MODEL_SERVER"
           value: "http://{{ include "rasa-x.host" $ }}:{{ $.Values.rasax.port }}/api/projects/default/models/tags/{{ .modelTag }}"
-
+        {{- include "rasa.extra.envs" $ | nindent 8 }}
```
Comment (wochinge) [inline charts/rasa-x/templates/rasa-deployments.yaml]: I think we should have separate `extraEnvs` for `rasa` and `rasa-x`.Their environment variables might be relatively similar now, but they might diverge more in the future.

File: charts/rasa-x/values.yaml (hunk 1 above)
Comment (whitedr) [reply charts/rasa-x/values.yaml]: It is intended to be a dict and yes I agree that a commented out example would make sense here.

File: charts/rasa-x/values.yaml (hunk 2 above)
Comment (whitedr) [reply charts/rasa-x/values.yaml]: Agree with the suggestion of following best practices here

File: charts/rasa-x/templates/rasa-deployments.yaml (hunk 3 above)
Comment (whitedr) [reply charts/rasa-x/templates/rasa-deployments.yaml]: Yes.  I thought of this and an initial version had them as separate.  I collapsed them as a simplification but I agree it's probably better in the long run for them to be allowed to be defined as a separate set of extra env vars.

File: charts/rasa-x/templates/rasa-x-deployment.yaml [hunk 4]
```diff
@@ -73,8 +73,6 @@ spec:
           value: "0"
         - name: "SANIC_RESPONSE_TIMEOUT"
           value: "3600"
-        - name: "LOCAL_MODE"
```
Comment (whitedr) [inline charts/rasa-x/templates/rasa-x-deployment.yaml]: Removed this as it was a duplicate which was defined above.

File: charts/rasa-x/templates/_helpers.tpl [hunk 5]
```diff
@@ -131,3 +131,21 @@ Return the port of the action container.
 {{- define "rasa-x.custom-actions.port" -}}
 {{- default 5055 .Values.app.port -}}
 {{- end -}}
+
```
Comment (whitedr) [inline charts/rasa-x/templates/_helpers.tpl]: These two defines allow for setting different extra environment variables for the rasa/rasa-x deployments.  They are used in `rasa-deployments.yaml` and `rasa-x-deployment.yaml` respectively and there are examples of defining extra env vars in the `values.yaml` file.

**Comment by whitedr** [comment]
@wochinge I've pushed some changes to this PR to hopefully address your comments/suggestions.  Let me know if it looks ok to you.  Thanks!

File: charts/rasa-x/templates/rasa-x-deployment.yaml (hunk 4 above)
Comment (wochinge) [reply charts/rasa-x/templates/rasa-x-deployment.yaml]: 🚀

**Comment by wochinge** [comment]
@whitedr Seems there is a merge conflict. Could you please resolve it? And please bump the chart version (1.1.1)

File: charts/rasa-x/templates/_helpers.tpl [hunk 6]
```diff
@@ -131,3 +131,21 @@ Return the port of the action container.
...
+{{/*
+Include rasa extra env vars
```
Comment (wochinge) [inline charts/rasa-x/templates/_helpers.tpl]:
```suggestion
Include rasa extra env vars.
```

File: charts/rasa-x/templates/_helpers.tpl [hunk 7]
```diff
@@ -131,3 +131,21 @@ Return the port of the action container.
...
+*/}}
+{{- define "rasa.extra.envs" -}}
+  {{- if .Values.rasa.extraEnvs -}}
+{{ toYaml .Values.rasa.extraEnvs }}
+  {{- end -}}
+{{- end -}}
+
+{{/*
+Include rasax extra env vars
```
Comment (wochinge) [inline charts/rasa-x/templates/_helpers.tpl]:
```suggestion
Include rasax extra env vars.
```

File: charts/rasa-x/values.yaml [hunk 8]
```diff
@@ -67,6 +71,19 @@ rasa:
   initialProbeDelay: 10
   # lockStoreDatabase is the database in redis which Rasa uses to store the conversation locks
   lockStoreDatabase: "1"
+  # extraEnvs are environment variables which can be added to the Rasa deployment
+  extraEnvs:
+    # example which sets env variables in rasa service from a separate k8s secret
```
Comment (wochinge) [inline charts/rasa-x/values.yaml]:
```suggestion
    # example which sets env variables in each Rasa Open Source service from a separate k8s secret
```

**Comment by wochinge** [comment]
resolved the merge conflict for you 🙂

**Pull request merged**